Framework/PythonInterface/plugins/algorithms/ConjoinSpectra.py

```diff
@@ -92,7 +92,7 @@ def PyExec(self):
             ta.setLabel(loopIndex, labelString)
             loopIndex += 1
             if mtd.doesExist(wsOutput):
-                ConjoinWorkspaces(InputWorkspace1=wsOutput, InputWorkspace2=wsTemp, CheckOverlapping=False)
```

Comment:
I can see there are lots of places where you now need to include `CheckMatchingBins=False`, why do these cases not result in a crash when the original issue does?

Reply:
The crash was being caused by `SaveNexusProcessed` and not by the `ConjoinWorkspaces` step mentioned in the original issue. The crash can still be caused by running the following:

```
w1=Load("SANS2D00089445.nxs",LoadMonitors=True)
w2=FilterByLogValue('w1','Julabo_1_ext',50,55)
w3=Rebin(w2,'0.0,100.0,10000.0',PreserveEvents=False)
w4=1.0*mtd['w1_monitors']
ConjoinWorkspaces('w4','w3',CheckMatchingBins=False) <- Updated line
SaveNexusProcessed('w4',"SANS2D00089445-temp6.nxs")
```

So this PR makes it difficult to crash but it does not completely eliminate it. For that probably another check on `SaveNexusProcessed` might be required down the line.

I asked Richard whether workspaces with mismatched bins will be conjoined or not. He said that it is a very rare use case. So that is why both options are available. However, the default behaviour is not allowing to conjoin if the bins are mismatched. 

Even though it is a "rare" use case, the doc example and many tests were conjoining workspaces with mismatched bins. So, I updated the tests by adding `CheckMatchingBins=False` to skip the check on the failing tests. 14 unit tests and 21 system tests were affected. 

The alternative was to `Rebin` in the tests before conjoining but this would have changed the subsequent values that were being asserted. For now, I have presumed that the tests where I have added `CheckMatchingBins=False` to `ConjoinWorkspaces` are testing for other things besides the behaviour of this function. So, I have tried to minimally change the tests. And adding the additional argument seemed like the best way to do it.